### PR TITLE
feat(run/receive-auth): add samples for receive-auth func as same python one

### LINKED ISSUE #2986: Sample Request: Receive authenticated requests
Request for a Go sample on [this documentation page](https://cloud.google.com/run/docs/authenticating/service-to-service#receive-request).

based on captured user feedback b/278311493

Only Python sample available: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/HEAD/run/service-auth/receive.py

### PR DESCRIPTION
**Pull request by ryuji-cre8ive**
## Description
* check jwt with email from the request.
* almost same func as python one
  * https://github.com/GoogleCloudPlatform/python-docs-samples/blob/HEAD/run/service-auth/receive.py 

Fixes #2986
Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [x] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [x] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [x] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [x] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [x] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [x] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: run/service-auth/receive.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		return
+	}
+
+	// split the auth type and value from the header.
+	authValues := strings.SplitN(authHeader, " ", 2)
+	if len(authValues) != 2 {
+		fmt.Fprintf(w, "Unhandled header format (%v).\n", authHeader)
+		return
+	}
+	authType, creds := authValues[0], authValues[1]
```
Comment (telpirion) [inline run/service-auth/receive.go:60]: nit: put a newline before this to improve readability.

File: run/service-auth/receive.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	if authType == "Bearer" {
+		token, err := jwt.Parse(creds, func(token *jwt.Token) (interface{}, error) {
```
Comment (telpirion) [inline run/service-auth/receive.go:62]: issue: run `gofmt` to ensure that this line is properly formatted. This looks a bit jumbled up and hard to read.

File: run/service-auth/receive.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+			}
+			// TODO: Set secret key
+			return []byte("my-secret-key"), nil
+		})
+		
+		if err != nil {
+			fmt.Fprintf(w, "Unable to parse token: %v", err)
+			return
+		}
+		claims, ok := token.Claims.(jwt.MapClaims)
```
Comment (telpirion) [inline run/service-auth/receive.go:75]: nit: insert newline before this line to improve readability.

File: run/service-auth/receive.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+		if !ok {
+			fmt.Fprintf(w, "Unable to extract claims from token")
+			return
+		}
+		email, ok := claims["email"].(string)
```
Comment (telpirion) [inline run/service-auth/receive.go:80]: nit: insert newline before this line to improve readability.

File: run/service-auth/receive.go [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+		if !ok {
+			fmt.Fprintf(w, "Unable to extract email from token")
+			return
+		}
+		fmt.Fprintf(w, "Hello, %v!\n", email)
+	} else {
```
Comment (telpirion) [inline run/service-auth/receive.go:86]: issue: rewrite this without the `else` clause.  Add a new check before line 61 like this:

```go
if authType != "Bearer" {
        fmt.Fprintf(w, "Unhandled header format (%v).\n", authType)
}
```

**Review by telpirion — CHANGES_REQUESTED** [review]

**Comment by ryuji-cre8ive** [comment]
How can I correct the error of `snippet-bot check — snippet-bot check`
should tags be 
```
[START cloudrun_<any-name-you-want>_dockerfile]
[START run_<any-name-you-want>_dockerfile] 
```


, shouldn't it?
that's why I named 
```
[START cloudrun_serviceauth_dockerfile]
[START run_serviceauth_dockerfile]
```

Thank you

File: run/service-auth/receive.go (hunk 1 above)
Comment (telpirion) [reply run/service-auth/receive.go:58]: Please don't resolve unaddressed comments without at least an "ack" and a reason why you haven't accepted the feedback.

File: run/service-auth/receive.go (hunk 2 above)
Comment (telpirion) [reply run/service-auth/receive.go:60]: See previous comment about providing a reason why comments are unaddressed.

File: run/service-auth/receive.go (hunk 3 above)
Comment (telpirion) [reply run/service-auth/receive.go:74]: See previous comment about providing a reason why comments are unaddressed.

File: run/service-auth/receive.go (hunk 4 above)
Comment (telpirion) [reply run/service-auth/receive.go:80]: See previous comment about providing a reason why comments are unaddressed.

**Review by telpirion — CHANGES_REQUESTED** [review]

**Comment by telpirion** [comment]
Please be sure that you are [following code review best practices](https://google.github.io/eng-practices/review/developer/handling-comments.html#code) with your pull requests.

**Comment by telpirion** [comment]
> How can I solve the error of `snippet-bot check — snippet-bot check` should tags be
> 
> ```
> [START cloudrun_<any-name-you-want>_dockerfile]
> [START run_<any-name-you-want>_dockerfile] 
> ```
> 
> , shouldn't it? that's why I named
> 
> ```
> [START cloudrun_serviceauth_dockerfile]
> [START run_serviceauth_dockerfile]
> ```
> 
> Thank you

It looks like the correct region tag prefix is `cloudrun`.

**Comment by telpirion** [comment]
Author hasn't addressed comments and this sample hasn't been planned for. Closing this PR.

**Pull request closed without merging**